pkg/infoschema/infoschema_v2_test.go

```diff
@@ -657,15 +657,34 @@ func TestDataStructFieldsCorrectnessInSchemaChange(t *testing.T) {
 	// verify table and partition related fields after drop table
 	txn, err = r.Store().Begin()
 	require.NoError(t, err)
-	_, err = builder.ApplyDiff(meta.NewMeta(txn), &model.SchemaDiff{Type: model.ActionDropTable, Version: 5, SchemaID: dbInfo.ID, TableID: tblInfo.ID})
+	m := meta.NewMeta(txn)
+	_, err = builder.ApplyDiff(m, &model.SchemaDiff{Type: model.ActionDropTable, Version: 5, SchemaID: dbInfo.ID, TableID: tblInfo.ID})
+	require.NoError(t, err)
+	// at first, the table will not be removed
+	tblItem, ok = v2.Data.byName.Get(tableItem{dbName: dbInfo.Name, tableName: tblInfo.Name, schemaVersion: 5})
+	require.True(t, ok)
+	require.False(t, tblItem.tomb)
+	tblItem, ok = v2.Data.byID.Get(tableItem{tableID: tblInfo.ID, schemaVersion: 5})
+	require.True(t, ok)
+	require.False(t, tblItem.tomb)
+	_, ok = v2.Data.tableCache.Get(tableCacheKey{tableID: tblInfo.ID, schemaVersion: 5})
+	require.True(t, ok)
+	require.Equal(t, v2.Data.pid2tid.Len(), 5) // tomb partition info
+	tblInfoItem, ok = v2.Data.pid2tid.Get(partitionItem{partitionID: 1, schemaVersion: 5})
+	require.True(t, ok)
+	require.False(t, tblInfoItem.tomb)
```

Comment:
Do you mean it's a drop table but the state is not public still, so not tomb?

Reply:
Yes.
The expected behavior is that all functions should return all tables in any state (except tables in `None` state), then the callers can filter out the needed tables.

If it's marked as `tomb`, it seems that some functions (like `IterateAllTableItems` and `SchemaSimpleTableInfos`) will not return the table, then it'll be easy for the callers to make mistakes. For example, during `drop schema`, tidb will call `checkDatabaseHasForeignKeyReferred -> is.SchemaSimpleTableInfos` to check whether a database has foreign key referred. If a table is marked `tomb`, it'll give unexpected result.

Though it's actually not a big issue because everything will finally be dropped, it's only an example to show that it's easy to make mistakes / the logic becomes hard to think or understand if it's marked as `tomb` during dropping tables.